blocks/v2-tabbed-carousel/v2-tabbed-carousel.js

```diff
@@ -0,0 +1,157 @@
+import { createElement, removeEmptyTags } from '../../scripts/common.js';
+
+const blockName = 'v2-tabbed-carousel';
+
+function buildTabNavigation(tabItems, clickHandler) {
+  const tabNavigation = createElement('ul', { classes: `${blockName}__navigation` });
+  const navigationLine = createElement('li', { classes: `${blockName}__navigation-line` });
+  let timeout;
+
+  [...tabItems].forEach((tabItem, i) => {
+    const listItem = createElement('li', { classes: `${blockName}__navigation-item` });
+    const button = createElement('button');
+    button.addEventListener('click', () => clickHandler(i));
+    if (navigationLine) {
+      button.addEventListener('mouseover', (e) => {
+        clearTimeout(timeout);
+        const { x: navigationX } = tabNavigation.getBoundingClientRect();
+        const { x, width } = e.currentTarget.getBoundingClientRect();
+        Object.assign(navigationLine.style, {
+          left: `${x + tabNavigation.scrollLeft - navigationX}px`,
+          width: `${width}px`,
+        });
+      });
+
+      button.addEventListener('mouseout', () => {
+        timeout = setTimeout(() => {
+          const { x: navigationX } = tabNavigation.getBoundingClientRect();
```

Comment:
We have a very similar code in the `mouseover` handler. Can we move it to a separate function like:
```
const navigatioLineRestyle = (item, item2) => { // have no idea how to call the arguments better
        const { x: navigationX } = item2.getBoundingClientRect();
        const { x, width } = item.getBoundingClientRect();
        Object.assign(navigationLine.style, {
          left: `${x + tabNavigation.scrollLeft - navigationX}px`,
          width: `${width}px`,
        });
}

button.addEventListener('mouseover', (e) => {
        clearTimeout(timeout);
        navigatioLineRestyle(e.target, tabNavigation);
}

button.addEventListener('mouseout', (e) => {
    timeout = setTimout(() => {
              const activeItem = document.querySelector(`.${blockName}__navigation-item.active`);
        navigatioLineRestyle(activeItem, tabNavigation)
    });
}
```